src/aggregate/functions/date.c

```diff
@@ -60,19 +60,20 @@ static int timeFormat(ExprEval *ctx, RSValue *result, RSValue **argv, size_t arg
 
 /* Fast alternative to mktime which is dog slow. From:
 https://gmbabar.wordpress.com/2010/12/01/mktime-slow-use-custom-function/ */
+// Fix and performance improvements:
+// https://godbolt.org/z/qscb5d9dT
+// https://quick-bench.com/q/oTV4_9uVqPTcrj2fpDEvbbMzZ48
+// https://quick-bench.com/q/2Bc8WY1Ys0vmbp-HPagWFxu81jI
 time_t fast_timegm(const struct tm *ltm) {
-  // elapsed days until the beginning of every month
-  const int mon_days[] = {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334};
-  long tyears, tdays, leaps, utc_hrs;
+  long tyears, tdays, leaps;
 
   tyears = ltm->tm_year - 70;  // tm->tm_year is from 1900.
```

Comment:
can we use fewer magic numbers? Sure, 365 is understandable. But 70 requires a comment, so it should probably be a `static const`. And 86400 (24\*60\*60) is not trivial to parse.

Reply:
70 got a comment (I can add `... and epoch is from 1970`)
86400 and 3600 I can replace them with `(24*60*60)` and `(60*60)`, the compiler should be able to produce the same binaries